CIP-?/README.md

```diff
@@ -0,0 +1,783 @@
+*mismatched length error* result, its error message must contain at least the 
+following information:
+
+- The name of the failed operation;
+- The reason (mismatched lengths); and
+- The lengths of the arguments.
+
+We describe the semantics of `andByteString`. For inputs $s$ and $t$, if
+$n \neq m$, the result is a mismatched length error. Otherwise, the result is 
+the byte sequence $u = u_n u_{n - 1} \ldots, u_0$ such that for all 
+$i \in \\{0, 1, \ldots, n\\}$ we have $u_i = 1$ if $s_i = t_i = 1$, and $0$
+otherwise.
+
+For `iorByteString`, for inputs $s$ and $t$, if $n \neq m$, the result is a 
+mismatched length error. Otherwise, the result is the byte sequence 
+$u = u_n u_{n - 1} \ldots u_0$ such that for all $i \in \\{0, 1, \ldots, n\\}$ 
+we have $u_i = 1$ if at least one of $s_i, t_i$ is $1$, and $0$ otherwise.
+
+For `xorByteString`, for inputs $s$ and $t$, if $n \neq m$, the result is a 
+mismatched length error. Otherwise, the result is the byte sequence 
+$u = u_n u_{n-1} \ldots u_0$ such that for all $i \in \\{0, 1, \ldots, n\\}$ 
+we have $u_i = 0$ if $s_i = t_i$, and $1$ otherwise.
+
+We observe that, for length-matched arguments, each of `andByteString`,
+`iorByteString` and `xorByteString` describes a commutative and associative 
+operation. Furthermore, for any given length $k$, each of these operations has 
+an identity element: for `iorByteString`, this is the bit sequence of length 
+$k$ where each element is $0$, and for `andByteString` and `xorByteString`, 
+this is the bit sequence of length $k$ where each element is $1$. Lastly, for 
+any length $k$, the bit sequence of length $k$ where each element is $0$ is an 
+absorbing element for `andByteString`, and the bit sequence of length $k$ 
+where each element is $1$ is an absorbing element for `iorByteString`.
+
+We now describe the semantics of `complementByteString`. For input $s$,
+the result is the byte sequence $u = u_n u_{n - 1} \ldots u_0$ such that for all
+$i \in \{0, 1, \ldots, n\}$ we have $u_i = 0$ if $s_i = 1$, and $1$ otherwise.
+
+We observe that `complementByteString` is self-inverting. We also note
+the following equivalences hold assuming `b` and `b'` have the
+same length; these are the DeMorgan laws:
+
+```haskell
+complementByteString (andByteString b b') = iorByteString (complementByteString b) (complementByteString b')
+```
+
+```haskell
+complementByteString (iorByteString b b') = andByteString (complementByteString b) (complementByteString b')
+```
+
+### Mixed operations
+
+Throughout this section, let $s = s_n s_{n-1} \ldots s_0$ be a byte sequence, 
+and let $i \in \mathbb{Z}$.
+
+We describe the semantics of `shiftByteString` and `rotateByteString`.
+Informally, both of these are 'index modifiers' for bit sequences: given a
+positive $i$, the index of a bit in $s$ 'increases' in the result; given a
+negative $i$, the index of a bit in $s$ 'decreases' in the result. This can mean
```

Comment:
_Now_ you specify it. I just think this should be in the brief description too.

Reply:
What would be a convenient way to briefly phrase this?

Reply:
Now done.